dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl

```diff
@@ -170,8 +176,6 @@ RUN readlink -f {{ $beatBinary }} | xargs setcap {{ .linux_capabilities }}
 RUN mkdir /app
 {{- end }}
 {{- else }}
```

Comment:
We may want to remove all this conditional logic depending on the user of the image, and remove the parts run only for user root.
This comes from the template used by Beats, where there were beats images run with user root (auditbeat and packetbeat IIRC). But all elastic-agent images use the `elastic-agent` user, find the definition [here](https://github.com/elastic/elastic-agent/blob/6e94de6503dae72637ce07d41635a5137acbb196/dev-tools/packaging/packages.yml#L659).
This would simplify this template.